### ISSUE TITLE
SSL client auth problem in version 5.5.1

### ISSUE THREAD

**Issue by sevdog** [elastic/elasticsearch-py#697]
With version `5.5.1` it is not possible to use SSL client auth (`client_cert` and `client_key`) when connection with `verify_certs=False`.

As [we can see in the code](https://github.com/elastic/elasticsearch-py/blob/5.5.1/elasticsearch/connection/http_urllib3.py#L76) if  we set  `verify_certs=False` then ssl client auth will be ignored. 

Also this behaviuor is undocumented, see [`5.5.1` docs](http://elasticsearch-py.readthedocs.io/en/5.5.1/connection.html#urllib3httpconnection-default-connection-class).

But in version `6.0.0` this configuration is possible ([see code](https://github.com/elastic/elasticsearch-py/blob/6.0.0/elasticsearch/connection/http_urllib3.py#L107) ).

The changes was introduced in `6.0.0` with [this commit](https://github.com/elastic/elasticsearch-py/commit/7339a272b58536921cc6a4f50d0aa12c67483566#diff-e03df718a08bdad9d78d94c50dd7db90).

Nevertheless the behaviour introduced in `6.0.0` seems correct, or at least removes an undocumented feature and permit a more flexible usage of the library, and should be backported also `5.x` releases.

**Comment by fxdgear** [comment]
@sevdog 

Can you please give me an example sample of code you're trying to use to connect?
I assume you are not using the SSLContext? As the SSLContext doesn't require the use of `client_cert` and `client_key`.

the behavior of using `verify_certs` and `client_cert` and `client_key` are deprecated. And i want to discourage using them, infavor of using `SSLContext` objects.

**Comment by sevdog** [comment]
Hi @fxdgear, here is my code (simplified for example purpose):
```
ELK_CONN_SETTING = {
    'use_ssl': True,
   'verify_certs': False,
    'ca_certs': os.path.join(CERTS_DIR, 'HTTPS_servers.pem'),
    'client_cert': os.path.join(CERTS_DIR, 'host.pem'),
    'client_key': os.path.join(CERTS_DIR, 'host.key')
}

es = elasticsearch.Elasticsearch(
    [server.hostname],
    **ELK_CONN_SETTING
)
```
I am using this code with `elasticsearch-5.5.1`, to connect with an Elasticsearch server with version `5.6.4`.

However, reading the code of tag `5.5.1` or branch `5.5.x` of this library you will notice that If you set `verify_certs=False` the two parameters `client_cert` and `client_key` wont be added to `kw`, and so will be ignored:
```
if verify_certs:
    if not ca_certs:
        raise ImproperlyConfigured("Root certificates are missing for certificate "
            "validation. Either pass them in using the ca_certs parameter or "
            "install certifi to use it automatically.")

    kw.update({
        'cert_reqs': 'CERT_REQUIRED',
        'ca_certs': ca_certs,
        'cert_file': client_cert,
        'key_file': client_key,
    })
else:
   # NOR cert_file NOR key_file will be passed to kw
   # and they will just be ignored
    warnings.warn(
        'Connecting to %s using SSL with verify_certs=False is insecure.' % host)
```
So, in versione `5.5.1` and previous of `elastichsearch-py`, it is not possible to connect to an Elasticsearch server with a self-signed certificate using SSL client auth.

This could be fine for production environment, but may be a pain in a development environment.

**Comment by fxdgear** [comment]
Thanks, looking into it

**Comment by fxdgear** [comment]
@sevdog I pushed a new version of elasticsearch-py last week. Can you please try the new build? 

Thanks

**Comment by sevdog** [comment]
@fxdgear Which version are you referring? 

If you mean [`6.1.1`](https://github.com/elastic/elasticsearch-py/blob/master/Changelog.rst#611-2018-01-05) it should work, as I said earlier from version `6.0.0` what I am reporting should not be an issue.

The issue is in version **proir** to `6.0.0` (`5.5.1` and below) as it is mentioned also in the title of this issue. As now I do not see any new commit on [`5.5.x` branch](https://github.com/elastic/elasticsearch-py/commits/5.5.x).

> I am using this code with elasticsearch-5.5.1, to connect with an Elasticsearch server with version 5.6.4.

ATM I have to work with an Elasticsearch `5.6.4`, which should be supported by branch `5.5.x` of this library. And your docs says that this is the branch that I should use:

> For Elasticsearch 6.0 and later, use the major version 6 (6.x.y) of the library.
>
> **For Elasticsearch 5.0 and later, use the major version 5 (5.x.y) of the library.**
>
> For Elasticsearch 2.0 and later, use the major version 2 (2.x.y) of the library, and so on.

If you say that I can work with Elasticsearch 5.x with elasticsearch-py 6.x I will be fine, but the problem IMO will remain on version 5.5.x of elasticsearch-py.

**Comment by fxdgear** [comment]
@sevdog please review: https://github.com/elastic/elasticsearch-py/pull/711

**Comment by fxdgear** [comment]
fixed in d021d19
